variables.tf

```diff
@@ -0,0 +1,19 @@
+variable "aws_region" {
+  description = "The AWS region to deploy into (e.g. us-east-1)."
+  default = "us-east-1"
+}
+
+variable "aws_eips_per_region" {
+  type = "map"
+  description = "The number of elastic IPs to create per AWS region."
+  default = {}
```

Comment:
Does this create 0 by default?  Can you add a comment about what is happening when it is undefined.   Can we set the `default = {}` to the actual default if it is zeros or ones?

Reply:
I think a default of zeroes makes sense.

Reply:
I don't see this changing much, and this will only be used in the production workspace (yes?).  So may be it should be set to the production values.  I'm afraid of someone firing this up and having all the EIPs destroyed because it is set to `0,0,0,0`.

Reply:
That said, maybe we need the `skip_destroy = true` magic added.

Reply:
My vote is for all zeroes with `skip_destroy = true`.

Reply:
@felddy, this will also be used in our own workspaces- here is how mine is currently set up:
`aws_eips_per_region = {
  us-east-1 = 0,
  us-east-2 = 2,
  us-west-1 = 0,
  us-west-2 = 0
}`

Reply:
I'm ok with the all zeroes, iff we call out in the `README.md` that this terraform is a noop until that is configured.

Reply:
Wait, why do devs need to run this in their own workspace?  Can't we just use ad hoc EIPs that normally get assigned to NAT gateways?  I thought the purpose of this was to make more permanent  EIPs for long-term production use.  

Can we make that happen?

Reply:
@felddy's comment makes sense.

Reply:
Fair enough- I will work on making that happen.

Reply:
@felddy, the code in the cyhy_amis project has been updated so that non-Production workspaces get ad-hoc EIPs, while Production workspaces are required to use a previously-created elastic IP (either via the code in this repo or manually).